tests/test_validation_framework/test_validation.py

```diff
@@ -426,6 +426,146 @@ def test_ascat_ismn_validation_metadata(ascat_reader):
     nptest.assert_equal(sorted(network), sorted(network_should))
 
 
+def test_validation_with_averager(ascat_reader):
+    """
+    Test processing framework with averaging module. ASCAT and ISMN data are used here with no geographical
+    considerations (the lut is provided more upstream and contains this information already)
+    """
+    while hasattr(ascat_reader, 'cls'):
+        ascat_reader = ascat_reader.cls
+    # lookup table between the ascat and ismn points - not geographically correct
+    upscaling_lut = {
+        "ISMN": {
+            1814367: [(0, 102.1333, 33.8833),(1, 102.1333, 33.6666)],
+            1803695: [(2, -86.55, 34.783),(3, -97.083, 37.133),(4, -105.417, 34.25)],
+            1856312: [(5, -120.9675, 38.43003),(6, -120.78559, 38.14956),(7, -120.80639, 38.17353)]
+        }}
+    gpis = (1814367, 1803695, 1856312)
+    lons, lats = [], []
+    for gpi in gpis:
+        lon, lat = ascat_reader.grid.gpi2lonlat(gpi)
+        lons.append(lon)
+        lats.append(lat)
+
+    jobs = [(gpis, lons, lats)]
+
+    # Initialize ISMN reader
+    ismn_data_folder = os.path.join(
+        os.path.dirname(__file__),
+        "..",
+        "test-data",
+        "ismn",
+        "multinetwork",
+        "header_values",
+    )
+    ismn_reader = ISMN_Interface(ismn_data_folder)
+
+    # Create the variable ***save_path*** which is a string representing the
+    # path where the results will be saved. **DO NOT CHANGE** the name
+    # ***save_path*** because it will be searched during the parallel
+    # processing!
+
+    save_path = tempfile.mkdtemp()
+
+    # Create the validation object.
+
+    datasets = {
+        "ASCAT": {
+            "class": ascat_reader,
+            "columns": ["sm"],
+            "kwargs": {
+                "mask_frozen_prob": 80,
+                "mask_snow_prob": 80,
+                "mask_ssf": True,
+            }
+        },
+        "ISMN": {
+            "class": ismn_reader,
+            "columns": ["soil moisture"],
+        },
+    }
```

Comment:
This is also used exactly like this in a lot of tests, could you make this a fixture too?